dali/operators/geometry/mt_transform_attr.cc

```diff
@@ -127,11 +127,11 @@ void MTTransformAttr::ProcessMatrixArg(const OpSpec &spec, const ArgumentWorkspa
     if (static_cast<int>(mtx_.size()) != output_pt_dim_ * input_pt_dim_) {
       mtx_.resize(output_pt_dim_ * input_pt_dim_, 0);
       MakeDiagonalMatrix(mtx_, 1);
-      Repeat(per_sample_mtx_, mtx_, N);
-      if (is_fused) {
-        translation_.resize(output_pt_dim_, 0);
-        Repeat(per_sample_translation_, translation_, N);
-      }
+    }
+    Repeat(per_sample_mtx_, mtx_, N);
+    if (is_fused) {
+      translation_.resize(output_pt_dim_, 0);
+      Repeat(per_sample_translation_, translation_, N);
```

Comment:
I guess this warrants another function than Repeat. It's pretty useless to overwrite the entire array if we only need to fill a few samples and that only in a now unlikely case (this will get more likely with conditional execution).

```C++
  template <typename OutRange, typename InRange>
  void ResizeBatch(OutRange &&out, InRange &&fill, int new_batch_size) {
  ssize_t n = dali::size(fill);
  assert(dali::size(out) % n == 0);
  ssize_t old_batch_size = dali::size(out) / n;
  resize_if_possible(out, n * new_batch_size);
  for (ssize_t i = old_batch_size, k = i * n; i < new_batch_size; i++)
      for (ssize_t j = 0; j < n; j++, k++)
      out[k] = fill[j];
  }
```

```suggestion
    ResizeBatch(per_sample_mtx_, mtx_, N);
    if (is_fused) {
      translation_.resize(output_pt_dim_, 0);
      ResizeBatch(per_sample_translation_, translation_, N);
```